dozer-sink-clickhouse/src/lib.rs

```diff
@@ -0,0 +1,449 @@
+    OptionalU128(Option<u128>),
+    OptionalInt(Option<i64>),
+    OptionalI128(Option<i128>),
+    OptionalFloat(
+        #[cfg_attr(feature= "arbitrary", arbitrary(with = arbitrary_float))]
+        Option<OrderedFloat<f64>>,
+    ),
+    OptionalBoolean(Option<bool>),
+    OptionalString(Option<String>),
+    OptionalText(Option<String>),
+    OptionalBinary(#[serde(with = "serde_bytes")] Option<Vec<u8>>),
+    OptionalDecimal(Option<Decimal>),
+    OptionalTimestamp(Option<DateTime<FixedOffset>>),
+    OptionalDate(Option<NaiveDate>),
+    OptionalJson(
+        #[cfg_attr(feature= "arbitrary", arbitrary(with = arb_json::arbitrary_json))]
+        Option<JsonValue>,
+    ),
+    OptionalPoint(Option<DozerPoint>),
+    OptionalDuration(Option<DozerDuration>),
+    Null(Option<()>),
+}
+
+fn convert_field_to_ff(field: Field, nullable: bool) -> FieldWrapper {
+    if nullable {
+        match field {
+            Field::UInt(v) => FieldWrapper::OptionalUInt(Some(v)),
+            Field::U128(v) => FieldWrapper::OptionalU128(Some(v)),
+            Field::Int(v) => FieldWrapper::OptionalInt(Some(v)),
+            Field::I128(v) => FieldWrapper::OptionalI128(Some(v)),
+            Field::Float(v) => FieldWrapper::OptionalFloat(Some(v)),
+            Field::Boolean(v) => FieldWrapper::OptionalBoolean(Some(v)),
+            Field::String(v) => FieldWrapper::OptionalString(Some(v)),
+            Field::Text(v) => FieldWrapper::OptionalText(Some(v)),
+            Field::Binary(v) => FieldWrapper::OptionalBinary(Some(v)),
+            Field::Decimal(v) => FieldWrapper::OptionalDecimal(Some(v)),
+            Field::Timestamp(v) => FieldWrapper::OptionalTimestamp(Some(v)),
+            Field::Date(v) => FieldWrapper::OptionalDate(Some(v)),
+            Field::Json(v) => FieldWrapper::OptionalJson(Some(v)),
+            Field::Point(v) => FieldWrapper::OptionalPoint(Some(v)),
+            Field::Duration(v) => FieldWrapper::OptionalDuration(Some(v)),
+            Field::Null => FieldWrapper::Null(None),
+        }
+    } else {
+        match field {
+            Field::UInt(v) => FieldWrapper::UInt(v),
+            Field::U128(v) => FieldWrapper::U128(v),
+            Field::Int(v) => FieldWrapper::Int(v),
+            Field::I128(v) => FieldWrapper::I128(v),
+            Field::Float(v) => FieldWrapper::Float(v),
+            Field::Boolean(v) => FieldWrapper::Boolean(v),
+            Field::String(v) => FieldWrapper::String(v),
+            Field::Text(v) => FieldWrapper::Text(v),
+            Field::Binary(v) => FieldWrapper::Binary(v),
+            Field::Decimal(v) => FieldWrapper::Decimal(v),
+            Field::Timestamp(v) => FieldWrapper::Timestamp(v),
+            Field::Date(v) => FieldWrapper::Date(v),
+            Field::Json(v) => FieldWrapper::Json(v),
+            Field::Point(v) => FieldWrapper::Point(v),
+            Field::Duration(v) => FieldWrapper::Duration(v),
+            Field::Null => FieldWrapper::Null(None),
```

Comment:
If it's not nullable, we can't map to null right? Or is this for the case where there's a default value?

Reply:
Updated binary handling part. During creation, we will show a warning if the source column is nullable. Also, during the insertion of data, we will map `null` as `vec![]`